obspy/imaging/tests/test_waveform.py

```diff
@@ -11,63 +11,64 @@
 import pytest
 
 
+def _create_stream(starttime, endtime, sampling_rate):
+    """
+    Helper method to create a Stream object that can be used for testing
+    waveform plotting.
+
+    Takes the time frame of the Stream to be created and a sampling rate.
+    Any other header information will have to be adjusted on a case by case
+    basis. Please remember to use the same sampling rate for one Trace as
+    merging and plotting will not work otherwise.
+
+    This method will create a single sine curve to a first approximation
+    with superimposed 10 smaller sine curves on it.
+
+    :return: Stream object
+    """
+    time_delta = endtime - starttime
+    number_of_samples = int(time_delta * sampling_rate) + 1
+    # Calculate first sine wave.
+    curve = np.linspace(0, 2 * np.pi, number_of_samples // 2)
+    # Superimpose it with a smaller but shorter wavelength sine wave.
+    curve = np.sin(curve) + 0.2 * np.sin(10 * curve)
+    # To get a thick curve alternate between two curves.
+    data = np.empty(number_of_samples)
+    # Check if even number and adjust if necessary.
+    if number_of_samples % 2 == 0:
+        data[0::2] = curve
+        data[1::2] = curve + 0.2
+    else:
+        data[-1] = 0.0
+        data[0:-1][0::2] = curve
+        data[0:-1][1::2] = curve + 0.2
```

Comment:
I know this is just the original code, but isn't it basically:
```suggestion
    # To get a thick curve alternate between two curves.
    curve[1::2] += 0.2
    # Check if even number and adjust if necessary.
    if number_of_samples % 2 == 1:
        curve[-1] = 0.0
```
(and then `data == curve`.)

Reply:
looks like data is twice as long as curve and then curve gets squashed in twice there? in any case, not worth spending time on beautifying this piece of super old test code, I'd say. :)